aten/src/ATen/BatchingRegistrations.cpp

```diff
@@ -165,6 +165,17 @@ Tensor diagonal_batching_rule(const Tensor& self, int64_t offset, int64_t dim1,
   return self_physical.newLogicalFromPhysical(result);
 }
 
+template <typename... Args> Tensor unsupportedRandomOp(Args... args) {
+  TORCH_CHECK(false, "vmap: We do not yet support random operations. Please ",
+              "perform random operations outside of vmap as a workaround");
+}
+
+template <typename... Args> Tensor& unsupportedRandomOp_(Args... args) {
+  TORCH_CHECK(false, "vmap: We do not yet support random operations. Please ",
+              "perform random operations outside of vmap as a workaround");
+}
```

Comment:
You could do this with marginally less boilerplate by making another magical unsupported boxed kernel, similar to the named tensor not supported kernel. Then it wouldn't be necessary to get the correct type instantiations below. Up to you.

Reply:
I'll add that as a follow-up item to investigate. Getting the correct type instantiations is annoying (and potentially fragile if someone updates the schema) so the magical unsupported kernel sounds nice.